File: ruby/p8.rb
#!/usr/bin/ruby


contents = File.read('p8.txt') 

tracker=[]
highestProduct = 0

contents.each_char { |chr|
	tracker.unshift(chr.to_i)
	if tracker.length > 5
		tracker.pop
	end
	if tracker.length == 5
		product = 1
		tracker.each do |x|
			product *= x
		end
		highestProduct = [highestProduct,product].max
	end
}

puts highestProduct

File: python/p8.py
#!/usr/bin/python

# I think this is the one where you have to find the highest product of any 5 digits in a row in the list

f = file('../p8.txt')
txt = f.read()
f.close()

txt = txt.strip()


length=len(txt)

print max([reduce(lambda x,y: int(x)*int(y), txt[i-5:i]) for i in range(5,length)]) 

